test/Log/LevelFilterTest.php

```diff
@@ -0,0 +1,103 @@
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     OpusTest
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace OpusTest\Log;
+
+use Opus\Log\LevelFilter;
+
+class LevelFilterTest extends \PHPUnit_Framework_TestCase
+{
+    public function testSetLevel()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+        $filter->setLevel(\Zend_Log::INFO);
+
+        $infoEvent = ['priority' => \Zend_Log::INFO];
+        $debugEvent = ['priority' => \Zend_Log::DEBUG];
+
+        $this->assertTrue($filter->accept($infoEvent));
+        $this->assertFalse($filter->accept($debugEvent));
+    }
+
+    public function testSetLevelOnNullArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+        $filter->setLevel(null);
+
+        $emergEvent = ['priority' => \Zend_Log::EMERG];
+        $debugEvent = ['priority' => \Zend_Log::DEBUG];
+
+        $this->assertTrue($filter->accept($emergEvent));
+        $this->assertTrue($filter->accept($debugEvent));
+    }
+
+    public function testSetLevelOnNegativeArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $this->setExpectedException(\InvalidArgumentException::class, 'Priority should be of Integer type and cannot be negative');
+
+        $filter->setLevel(-1);
+    }
+
+    public function testSetLevelOnStringArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $this->setExpectedException(\InvalidArgumentException::class, 'Priority should be of Integer type and cannot be negative');
+
+        $filter->setLevel('TestPriority');
+    }
+
+    public function testGetLevel()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $this->assertEquals(\Zend_Log::WARN, $filter->getLevel());
+    }
+
+    public function testGetLevelForDisabledFilter()
+    {
+        $filter = new LevelFilter(\Zend_Log::EMERG, '>=');
```

Comment:
That is the weird case I was talking about a little further up. I know what you mean here, but really, isn't there a way to make this easier to understand?

Reply:
Done.